src/peripheral/nvic.rs

```diff
@@ -65,9 +65,30 @@ pub struct RegisterBlock {
     /// so convenient byte-sized representation wouldn't work on that
     /// architecture.
     pub ipr: [RW<u32>; 8],
+
+    #[cfg(not(armv6m))]
+    reserved5: [u32; 208],
+
+    #[cfg(armv6m)]
+    reserved5: [u32; 696],
+
+    /// Software Trigger Interrupt
+    pub stir: WO<u32>,
 }
 
 impl NVIC {
+    /// Request an IRQ in software
+    pub fn req_irq<I>(&mut self, interrupt: I)
```

Comment:
I think this method should be renamed to `request`. The other NVIC methods like `enabled(interrupt: I)` are read as "enable interrupt", so I think "request interrupt" would be more fitting.

I think this method should include a comment similar to what's included in the ARM documentation:

> Writing a value to the INTID field is the same as manually pending an interrupt by setting the corresponding interrupt bit in an Interrupt Set Pending Register.

Reply:
> I think this method should include a comment similar to what's included in the ARM documentation:

in particular the comment should mention that this method is similar to `set_pending`.